ocaml/sdk-gen/component-test/jsonrpc-client/go/error_test.go

```diff
@@ -0,0 +1,123 @@
+package componenttest
+
+import (
+	"errors"
+	"strings"
+	"testing"
+	"xenapi"
+)
+
+func TestRpcError(t *testing.T) {
+	//Constuct an rpc client error
+	data, err := ReadJsonFile("../../spec/xapi-24/rpc_error.json")
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	testId, err := GetTestId(data)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+	session, err := GetSession(testId)
+	if err != nil {
+		t.Log(err)
+		t.Fail()
+		return
+	}
+
+	var vmRef xenapi.VMRef = "OpaqueRef:6ef08bce-0bf0-30ff-804f-5f0ee4bbdd13"
```

Comment:
I think the assumption on the spec is that the methods in the `method` list is a set.
```
"method": [
        "VM.get_record"
      ],
```
This could be improved in the future. But this also means the tester knows there must be a `VM.get_record` in 
```
"params": {
        "VM.get_record": ["", "OpaqueRef:6ef08bce-0bf0-30ff-804f-5f0ee4bbdd13"]
      },
```
And the `vmRef` can be read from the spec in the current test context which is defined by the specific spec.